### PR TITLE
Project 19 - Johanna Åstrand

### PR DESCRIPTION
**Pull request by jastrand**
This is my project! Getting the likes to work was a bit tricky and I would have loved to spend some time to add more features but here it is: 

Frontend: https://infallible-archimedes-9a857b.netlify.app/
Backend: https://happy-thoughts-api-johanna.herokuapp.com/thoughts

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3,14 +3,32 @@ import bodyParser from 'body-parser'
...
+  message: {
+    type: String,
+    required: true,
+    minlenght: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+    required: true
```
Comment (KarolCislowski) [inline server.js:21]: If you defining the default field, there is no need to make it required, because it will never be empty :)

File: server.js [hunk 2]
```diff
@@ -23,6 +41,37 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  res.json(thoughts)
+})
```
Comment (KarolCislowski) [inline server.js:47]: according to documentation `get "/"` need to return 20 thoughts,

File: server.js [hunk 3]
```diff
@@ -23,6 +41,37 @@ app.get('/', (req, res) => {
...
+  const thought = new Thought({ message })
+
+  try {
+    //success
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  } catch (err) {
+    res.status(400).json({ message: 'could not save thought to the database', error: err.errors })
+  }
+})
```
Comment (KarolCislowski) [inline server.js:62]: according to documentation `post "/"` need to create a new one.

File: server.js [hunk 4]
```diff
@@ -23,6 +41,37 @@ app.get('/', (req, res) => {
...
+  try {
+    const savedLike = await Thought.updateOne(
+      { _id: req.params.id }, { $inc: { hearts: 1 } }
+    )
+    res.json(savedLike)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not like post, id not found', error: err.errors })
+  }
+})
+
```
Comment (KarolCislowski) [inline server.js:74]: This one should be `/:thoughtId/like`

File: server.js (hunk 1 above)
Comment (jastrand) [reply server.js:21]: True, will fix!

File: server.js (hunk 2 above)
Comment (jastrand) [reply server.js:47]: I dont really understand this point? It is returning max 20 thoughts?

File: server.js (hunk 3 above)
Comment (jastrand) [reply server.js:62]: Not sure I understand this either?

File: server.js (hunk 4 above)
Comment (jastrand) [reply server.js:74]: to make it more RESTful?

File: server.js (hunk 2 above)
Comment (KarolCislowski) [reply server.js:47]: You used ` get /thoughts` and the documentation forced you to use ` get /`

File: server.js (hunk 4 above)
Comment (KarolCislowski) [reply server.js:74]: To meet the requirements. https://www.notion.so/Project-Happy-Thoughts-API-3dc1710667304987894f954dad265fc9

**Pull request closed without merging**